bin/server.js

```diff
@@ -5,124 +5,151 @@ var debug       = require('debug')('index:bin:server');
 var base        = require('taskcluster-base');
 var taskcluster = require('taskcluster-client');
 var data        = require('../index/data');
+var Handlers    = require('../index/handlers');
 var v1          = require('../routes/api/v1');
+var loader      = require('taskcluster-lib-loader');
 
-/** Launch server */
-var launch = function(profile) {
-  // Load configuration
-  var cfg = base.config({
-    defaults:     require('../config/defaults'),
-    profile:      require('../config/' + profile),
-    envs: [
-      'index_publishMetaData',
-      'taskcluster_queueBaseUrl',
-      'taskcluster_authBaseUrl',
-      'taskcluster_credentials_clientId',
-      'taskcluster_credentials_accessToken',
-      'index_azureAccount',
-      'aws_accessKeyId',
-      'aws_secretAccessKey',
-      'influx_connectionString'
-    ],
-    filename:     'taskcluster-index'
-  });
+// Create component loader
+var load = loader({
+  cfg: {
+    requires: ['profile'],
+    setup: ({profile}) => base.config({profile}),
+  },
 
   // Configure IndexedTask and Namespace entities
-  var IndexedTask = data.IndexedTask.setup({
-    account:          cfg.get('index:azureAccount'),
-    table:            cfg.get('index:indexedTaskTableName'),
-    credentials:      cfg.get('taskcluster:credentials'),
-    authBaseUrl:      cfg.get('taskcluster:authBaseUrl')
-  });
-  var Namespace = data.Namespace.setup({
-    account:          cfg.get('index:azureAccount'),
-    table:            cfg.get('index:namespaceTableName'),
-    credentials:      cfg.get('taskcluster:credentials'),
-    authBaseUrl:      cfg.get('taskcluster:authBaseUrl')
-  });
+  IndexedTask: {
+    requires: ['cfg'],
+    setup: ({cfg}) => data.IndexedTask.setup({
+      account:          cfg.index.azureAccount,
+      table:            cfg.index.indexedTaskTableName,
+      credentials:      cfg.taskcluster.credentials,
+      authBaseUrl:      cfg.taskcluster.authBaseUrl
+    })
+  },
+  Namespace: {
+    requires: ['cfg'],
+    setup: ({cfg}) => data.Namespace.setup({
+      account:          cfg.index.azureAccount,
+      table:            cfg.index.namespaceTableName,
+      credentials:      cfg.taskcluster.credentials,
+      authBaseUrl:      cfg.taskcluster.authBaseUrl
+    })
+  },
 
   // Create a validator
-  var validator = null;
-  var validatorCreated = base.validator({
-    folder:           path.join(__dirname, '..', 'schemas'),
-    constants:        require('../schemas/constants'),
-    publish:          cfg.get('index:publishMetaData') === 'true',
-    schemaPrefix:     'index/v1/',
-    aws:              cfg.get('aws')
-  }).then(function(validator_) {
-    validator = validator_;
-  });
+  validator: {
+    requires: ['cfg'],
+    setup: ({cfg}) => base.validator({
+      folder:           path.join(__dirname, '..', 'schemas'),
+      constants:        require('../schemas/constants'),
+      publish:          cfg.index.publishMetaData === 'true',
```

Comment:
Let's use `cfg.index.publishMetaData` directly, we have booleans with `typed-env-config`... will add comment in config.yml